elucidator-examples/c/monte_carlo/monte_carlo.c

```diff
@@ -0,0 +1,166 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <time.h>
+#include <math.h>
+#include "elucidator.h"
+
+#define N_STEPS 100000
+#define SAMPLES_PER_STEP 50
+#define DISPLAY_INTERVAL 5000
+
+#define MADEIT() \
+    do { \
+        printf("Made it to line %d\n", __LINE__); \
+    } while (0)
+
+
+typedef struct {
+    uint64_t hits;
+    uint64_t misses;
+} StepSummary;
+
+ElucidatorStatus print_error_if_not_ok(ErrorHandle * eh, ElucidatorStatus status); 
+ElucidatorStatus setup(SessionHandle * sh, ErrorHandle * eh);
+ElucidatorStatus run_experiment(SessionHandle * sh, ErrorHandle * eh, uint64_t n_steps, uint64_t samples_per_step);
+StepSummary run_experiment_step(uint64_t samples_per_step);
+ElucidatorStatus run_analysis(SessionHandle * sh, ErrorHandle * eh, uint64_t n_steps, uint64_t display_interval);
+StepSummary summarize_bufnodes(BufNode **bufnodes);
+
+int main() {
+    srand(time(NULL));
+
+    SessionHandle * sh = ALLOCATE_HANDLE();
+    ErrorHandle * eh = ALLOCATE_HANDLE();
+    ElucidatorStatus status;
+
+    status = setup(sh, eh);
+    if (status != ELUCIDATOR_OK) {
+        fprintf(stderr, "Setup failed. Exiting.");
+        return 1;
+    }
+    run_experiment(sh, eh, N_STEPS, SAMPLES_PER_STEP);
+    if (status != ELUCIDATOR_OK) {
+        fprintf(stderr, "Experiment failed. Exiting.");
+        return 1;
+    }
+    run_analysis(sh, eh, N_STEPS, DISPLAY_INTERVAL);
+    if (status != ELUCIDATOR_OK) {
+        fprintf(stderr, "Analysis failed. Exiting.");
+        return 1;
+    }
+    free(sh);
+    free(eh);
+}
+
+
+ElucidatorStatus print_error_if_not_ok(ErrorHandle * eh, ElucidatorStatus status) {
+    if ( status != ELUCIDATOR_OK ) {
+        char * msg = get_error_string(eh);
+        fprintf(stderr, "%s\n", msg);
+        free(msg);
+    }
+    return status;
+}
+
+ElucidatorStatus setup(SessionHandle * sh, ErrorHandle * eh) {
+    ElucidatorStatus status;
+    char * designation = "state";
+    char * spec = "hits: u64, misses: u64";
+
+    status = new_session(sh, ELUCIDATOR_RTREE);
+    if (print_error_if_not_ok(eh, status) != ELUCIDATOR_OK) return status;
+    status = add_spec_to_session(designation, spec, sh, eh);
+    if (print_error_if_not_ok(eh, status) != ELUCIDATOR_OK) return status;
+    return status;
```

Comment:
```suggestion
    print_error_if_not_ok(eh, status);
    return status;
```
You're exiting either way afterwards, the conditional is redundant.